### PR TITLE
Upgrade to aws-sdk v3

### PR DESCRIPTION
**Pull request by iansu**
- Upgrade to aws-sdk v3
- Update dependencies
- Increase minimum Node version to 18 (current oldest LTS version)

### CODE REVIEW

File: src/get-secret.ts [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
 
 const getSecret = (secretName: string, region: string, timeout: number): SecretObject => {
   try {
-    const secret = awsParamStore.getParameterSync(`/aws/reference/secretsmanager/${secretName}`, {
-      region,
-      httpOptions: {
-        timeout,
-      },
-    });
+    const rawValue = spawnSync('node', [path.join(__dirname, 'secrets-manager.js')], {
```
Comment (ScottKayeNeo) [inline src/get-secret.ts:9]: Hahaha I love this

**Review by ScottKayeNeo — APPROVED** [review]
I will test this in a service later, but I remember the context behind this change, and the async-to-sync magic is great.

**Review by AD-Blue — APPROVED** [review]

**Review by AdamTranquilla — APPROVED** [review]

**Review by Lewis65 — DISMISSED** [review]
Thanks for doing this! 🤜

**Comment by ScottKayeNeo** [comment]
Currently letting this sit in cardinal-service with version `config-dug@1.8.0-next.0`

File: tsconfig.build.json [hunk 2]
```diff
@@ -1,13 +1,9 @@
...
+  "extends": "@tsconfig/node18",
   "compilerOptions": {
-    "module": "commonjs",
-    "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
-    "target": "es6",
-    "noImplicitAny": true,
-    "moduleResolution": "node",
-    "sourceMap": true,
-    "declaration": true,
```
Comment (ScottKayeNeo) [inline tsconfig.build.json:10]: We need to keep `declaration: true`, otherwise the build doesn't contain type information and TypeScript will complain :/

Comment (ScottKayeNeo) [reply tsconfig.build.json:10]: Testing this now with `1.8.0-next.1`

**Review by ScottKayeNeo — CHANGES_REQUESTED** [review]
Requesting changes just because of the missing `declaration` setting in the build config - this appears to break the build because there isn't any type information to verify against.  The classic:

```
....ts:2:20 - error TS7016: Could not find a declaration file for module 'config-dug'. '...' implicitly has an 'any' type.
  Try `npm i --save-dev @types/config-dug` if it exists or add a new declaration (.d.ts) file containing `declare module 'config-dug';`

2 import config from 'config-dug';
                     ~~~~~~~~~~~~
```

**Comment by iansu** [comment]
@ScottKayeNeo I fixed the types issue and published `1.8.0-next.2`

**Review by ScottKayeNeo — APPROVED** [review]

**Review by nozomu-koshirae — APPROVED** [review]
No issues so far with OLS running `1.8.0-next.2`  either!

**Pull request merged**